canard.c

```diff
@@ -131,6 +131,10 @@ uint8_t canardGetLocalNodeID(const CanardInstance* ins)
     return ins->node_id;
 }
 
+void canardForgetLocalNodeID(CanardInstance* ins) {
+    ins->node_id = CANARD_BROADCAST_NODE_ID;
```

Comment:
Just a quick concern I have about this specific branch.

For any CanardInstance all the following are being tracked:

https://github.com/matternet/libcanard/blob/d1d7230ab67048da60083465eda3cc818ddd1b90/canard.h#L258

`struct CanardInstance
{
    uint8_t node_id;                                ///< Local node ID; may be zero if the node is anonymous

    CanardShouldAcceptTransfer should_accept;       ///< Function to decide whether the application wants this transfer
    CanardOnTransferReception on_reception;         ///< Function the library calls after RX transfer is complete

    CanardPoolAllocator allocator;                  ///< Pool allocator

    CanardRxState* rx_states;                       ///< RX transfer states
    CanardTxQueueItem* tx_queue;                    ///< TX frames awaiting transmission

    void* user_reference;                           ///< User pointer that can link this instance with other objects
};`

I'm not at all familiar with canard.  Are we sure that this is all we need to do to is set node_id = CANARD_BROADCAST_NODE_ID?  

These changes are in a branch and after looking through all the call stacks this code isn't getting used by station firmwares AFAIK.  I personally do not know how well the changes that are on this branch have been tested and validated or if at all.  Would it be safer and cleaner not to include this branch (and the associated framework branch) into main?

Reply:
@kuanli1 This function is being called in bmu_firmware, particularly when the BMU goes to sleep after being disconnected (from either aircraft or charger).  This call has been in place since Mitch committed it in March 2018 with commit e1a21aa. 
This change is in our BMU releases since v2.4.  I don't know what formal testing was done for this.  However, I have taken the current BMU firmware in and out of sleep and have not experienced any issues with neither going to sleep nor recovering from sleep and waking up.

In short, this change isn't needed in station_firmwares but it is needed in bmu_firmware.  Given the fact that all of our framework-based software (i.e. bmu_firmware and station_firmwares) already have this commit integrated (via libcanard branch forget-uavcan-nodeid), I think it's safe to merge into master and we all work with master from hereon.

Thoughts?

Reply:
I'm ok with merging this into the main branch with the information you provided.